src/models/application-membership.ts

```diff
@@ -53,7 +53,7 @@ const getApplicationMembershipModel = function (
 		const role = await pine.get({
 			resource: 'application_membership_role',
 			id: {
-				name: roleName,
+				name: roleName as ApplicationMembershipRoles,
```

Comment:
These 2 casts are necessary because the type being passed in is `string` but that doesn't match the custom typing meaning the resource lookup fails and the returned `id` becomes `string | number | undefined` rather than `number | undefined` which then fails later as it passes the id to a model that doesn't allow a string. Doing this cast means it can resolve the resource properly and continue to infer the `number | undefined` response